table_schema_cache.go

```diff
@@ -211,6 +209,70 @@ func LoadTables(db *sql.DB, tableFilter TableFilter, columnCompressionConfig Col
 	return tableSchemaCache, nil
 }
 
+func (ts *TableSchema) findColumnByName(name string) *schema.TableColumn {
+	for _, column := range ts.Columns {
+		if column.Name == name {
+			return &column
+		}
+	}
+	return nil
+}
+
+func LackPaginationKeyError(schema, table string, hasCompositePrimaryKey bool) error {
+	if hasCompositePrimaryKey {
+		return fmt.Errorf("no Pagination Key Column provided for %s.%s which has a composite primary key. Attempt to fallback to `id` column failed as the same is not existent. Kindly specify a Pagination Key Column", schema, table)
+	}
+	return fmt.Errorf("no Pagination Key Column provided for %s.%s which has no Primary Key to fallback to. Attempt to fallback to `id` column failed as the same is not existent. Kindly specify a Pagination Key Column", schema, table)
+}
+
+func PaginationKeyNotFoundError(schema, table, paginationColumn string) error {
+	return fmt.Errorf("Pagination Key Column provided: %s for %s.%s is a non existing column", paginationColumn, schema, table)
+}
+
+func (ts *TableSchema) setPaginationKeyColumn(dbname, table string, paginateByColumnConfig PaginateByColumnConfig) error {
+	var err error
+	if paginationColumn, found := paginateByColumnConfig.PaginationColumnFor(dbname, table); found {
+		ts.PaginationKeyColumn = ts.findColumnByName(paginationColumn)
+		err = PaginationKeyNotFoundError(dbname, table, paginationColumn)
+	} else {
+		switch len(ts.PKColumns) {
+		case 0:
+			ts.PaginationKeyColumn = ts.findColumnByName("id")
+			err = LackPaginationKeyError(dbname, table, false)
+		case 1:
+			ts.PaginationKeyColumn = &ts.Columns[ts.PKColumns[0]]
+		default:
+			ts.PaginationKeyColumn = ts.findColumnByName("id")
+			err = LackPaginationKeyError(dbname, table, true)
+		}
+	}
+	if ts.PaginationKeyColumn == nil {
+		return err
+	}
+
+	return nil
```

Comment:
This whole method is a bit weird.. I'd prefer if it didn't set `ts.PaginationKeyColumn` directly but only inferred what it should be based on configuration. i.e. lift up the field assignment one level up.

I would then rewrite it as something like:

```
func (ts *TableSchema) setPaginationKeyColumn(dbname, table string, paginateByColumnConfig PaginateByColumnConfig) (*schema.TableColumn, error) {
    if paginateByColumnConfig == nil {
        return inferPaginationColumn(ts, dbname, table)
    } else {
        // ...
    } 
```

Reply:
Will address this as soon as the above discussion is resolved.